src/Frontend/MainWindow.vala

```diff
@@ -69,52 +70,66 @@ class Taxi.MainWindow : Hdy.ApplicationWindow {
+            transition_type = Gtk.RevealerTransitionType.SLIDE_RIGHT,
+            child = operations_button
+        };
 
-        bookmark_list = new Gtk.Grid ();
-        bookmark_list.margin_top = bookmark_list.margin_bottom = 3;
-        bookmark_list.orientation = Gtk.Orientation.VERTICAL;
+        bookmark_list = new Gtk.Box (Gtk.Orientation.VERTICAL, 0);
+        bookmark_list.add_css_class (Granite.STYLE_CLASS_MENU);
 
-        var bookmark_scrollbox = new Gtk.ScrolledWindow (null, null);
-        bookmark_scrollbox.hscrollbar_policy = Gtk.PolicyType.NEVER;
-        bookmark_scrollbox.max_content_height = 500;
-        bookmark_scrollbox.propagate_natural_height = true;
-        bookmark_scrollbox.add (bookmark_list);
-        bookmark_scrollbox.show ();
+        var bookmark_scrollbox = new Gtk.ScrolledWindow () {
+            child = bookmark_list,
+            hscrollbar_policy = Gtk.PolicyType.NEVER,
+            max_content_height = 500,
+            propagate_natural_height = true,
+        };
 
-        var bookmark_popover = new Gtk.Popover (null);
-        bookmark_popover.add (bookmark_scrollbox);
+        bookmark_popover = new Gtk.Popover () {
+            child = bookmark_scrollbox,
+            width_request = 250
+        };
 
-        bookmark_menu_button = new Gtk.MenuButton ();
-        bookmark_menu_button.image = new Gtk.Image.from_icon_name ("user-bookmarks", Gtk.IconSize.LARGE_TOOLBAR);
-        bookmark_menu_button.popover = bookmark_popover;
-        bookmark_menu_button.tooltip_text = _("Access Bookmarks");
+        bookmark_menu_button = new Gtk.MenuButton () {
+            icon_name = "user-bookmarks",
+            tooltip_text = _("Access Bookmarks"),
+            popover = bookmark_popover,
+            valign = CENTER
+        };
+        bookmark_menu_button.add_css_class (Granite.STYLE_CLASS_LARGE_ICONS);
 
         update_bookmark_menu ();
 
-        var header_bar = new Hdy.HeaderBar () {
-            custom_title = new Gtk.Label (null),
-            show_close_button = true
+        var header_bar = new Adw.HeaderBar () {
+            show_title = false
         };
         header_bar.pack_start (connect_box);
         header_bar.pack_start (spinner_revealer);
         header_bar.pack_start (bookmark_menu_button);
 
-        welcome = new Granite.Widgets.Welcome (
-            _("Connect"),
-            _("Type a URL and press 'Enter' to\nconnect to a server.")
-        );
-        welcome.vexpand = true;
+        var app_icon = new Gtk.Image.from_icon_name ("com.github.alecaddd.taxi") {
+            pixel_size = 64
+        };
+
+        title_label = new Gtk.Label (_("Connect"));
+        title_label.add_css_class (Granite.STYLE_CLASS_H1_LABEL);
+
+        var description_label = new Gtk.Label (_("Type a URL and press 'Enter' to\nconnect to a server.")) {
+            justify = CENTER
+        };
+
+        welcome_box = new Granite.Placeholder (_("Connect")) {
```

Comment:
I prefer to name this variable as `welcome` to match `Granite.Widgets.Welcome` in the previous code to (possibly) reduce the diff and plus `Granite.Placeholder` is not a Box